### PR TITLE
Lbank2 deposit withdraw fee

### PR DESCRIPTION
**Pull request by samgermain**
needs https://github.com/ccxt/ccxt/pull/15812 merged for the `networks` keys to be uppercase

---------------

# testing

## fetchPrivateDepositWithdrawFees

```
2022-11-24T23:19:49.889Z
Node.js: v18.4.0
CCXT v2.2.29
lbank2.fetchDepositWithdrawFees ()
2022-11-24T23:19:52.176Z iteration 0 passed in 441 ms

{
  USDT: {
    info: {
      usableAmt: '14.36',
      assetAmt: '14.36',
      networkList: [
        {
          isDefault: false,
          withdrawFeeRate: '',
          name: 'erc20',
          withdrawMin: '20',
          minLimit: '0.0001',
          minDeposit: '20',
          feeAssetCode: 'usdt',
          withdrawFee: '20',
          type: '1',
          coin: 'usdt',
          network: 'eth'
        },
        {
          isDefault: false,
          name: 'omni',
          coin: 'usdt',
          network: 'btc'
        },
        {
          isDefault: false,
          withdrawFeeRate: '',
          name: 'trc20',
          withdrawMin: '2',
          minLimit: '0.0001',
          minDeposit: '0.1',
          feeAssetCode: 'usdt',
          withdrawFee: '1',
          type: '1',
          coin: 'usdt',
          network: 'trx'
        },
        {
          isDefault: false,
          withdrawFeeRate: '',
          name: 'asa',
          withdrawMin: '1E+1',
          minLimit: '0.0001',
          minDeposit: '5',
          feeAssetCode: 'usdt',
          withdrawFee: '5',
          type: '1',
          coin: 'usdt',
          network: 'algo'
        },
        {
          isDefault: false,
          withdrawFeeRate: '',
          name: 'bep20(bsc)',
          withdrawMin: '10',
          minLimit: '0.0001',
          minDeposit: '0.0001',
          feeAssetCode: 'usdt',
          withdrawFee: '1',
          type: '1',
          coin: 'usdt',
          network: 'bsc'
        },
        {
          isDefault: false,
          withdrawFeeRate: '',
          name: 'heco',
          withdrawMin: '2',
          minLimit: '0.0001',
          minDeposit: '0.1',
          feeAssetCode: 'usdt',
          withdrawFee: '1',
          type: '1',
          coin: 'usdt',
          network: 'heco'
        },
        {
          isDefault: true,
          withdrawFeeRate: '',
          name: 'matic',
          withdrawMin: '1',
          minLimit: '0.0001',
          minDeposit: '1',
          feeAssetCode: 'usdt',
          withdrawFee: '1',
          type: '1',
          coin: 'usdt',
          network: 'matic'
        }
      ],
      freezeAmt: '0',
      coin: 'usdt'
    },
    withdraw: { fee: 1, percentage: undefined },
    deposit: { fee: undefined, percentage: undefined },
    networks: {
      ERC20: {
        withdraw: { fee: 20, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      },
      TRC20: {
        withdraw: { fee: 1, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      },
      ASA: {
        withdraw: { fee: 5, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      },
      BSC: {
        withdraw: { fee: 1, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      },
      HT: {
        withdraw: { fee: 1, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      },
      MATIC: {
        withdraw: { fee: 1, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      }
    }
  },
  ADA: {
    info: {
      usableAmt: '2.65064888',
      assetAmt: '2.65064888',
      networkList: [
        {
          isDefault: true,
          withdrawFeeRate: '',
          name: 'ada',
          withdrawMin: '5',
          minLimit: '0.0001',
          minDeposit: '2',
          feeAssetCode: 'ada',
          withdrawFee: '1',
          type: '1',
          coin: 'ada',
          network: 'ada'
        }
      ],
      freezeAmt: '0',
      coin: 'ada'
    },
    withdraw: { fee: 1, percentage: undefined },
    deposit: { fee: undefined, percentage: undefined },
    networks: {
      ada: {
        withdraw: { fee: 1, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      }
    }
  }
}
2022-11-24T23:19:52.176Z iteration 1 passed in 441 ms
```

## fetchPublicDepositWithdrawFees

```
2022-11-24T23:24:24.624Z
Node.js: v18.4.0
CCXT v2.2.29
lbank2.fetchDepositWithdrawFees ()
2022-11-24T23:24:27.075Z iteration 0 passed in 730 ms

{
  LBK: {
    info: [
      {
        amountScale: '4',
        chain: 'heco',
        assetCode: 'lbk',
        min: '200',
        transferAmtScale: '4',
        canWithDraw: true,
        fee: '100',
        minTransfer: '0.0001',
        type: '1'
      },
      {
        amountScale: '4',
        chain: 'erc20',
        assetCode: 'lbk',
        min: '4000',
        transferAmtScale: '4',
        canWithDraw: true,
        fee: '2000',
        minTransfer: '0.0001',
        type: '1'
      }
    ],
    withdraw: { fee: undefined, percentage: undefined },
    deposit: { fee: undefined, percentage: undefined },
    networks: {
      HT: {
        withdraw: { fee: 100, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      },
      ERC20: {
        withdraw: { fee: 2000, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      }
    }
  },
  USDT: {
    info: [
      {
        amountScale: '4',
        chain: 'erc20',
        assetCode: 'usdt',
        min: '60',
        transferAmtScale: '4',
        canWithDraw: true,
        fee: '30',
        minTransfer: '0.0001',
        type: '1'
      },
      {
        amountScale: '4',
        chain: 'trc20',
        assetCode: 'usdt',
        min: '2',
        transferAmtScale: '4',
        canWithDraw: true,
        fee: '1',
        minTransfer: '0.0001',
        type: '1'
      },
      {
        amountScale: '4',
        chain: 'bep20(bsc)',
        assetCode: 'usdt',
        min: '10',
        transferAmtScale: '4',
        canWithDraw: true,
        fee: '1',
        minTransfer: '0.0001',
        type: '1'
      },
      {
        amountScale: '4',
        chain: 'heco',
        assetCode: 'usdt',
        min: '2',
        transferAmtScale: '4',
        canWithDraw: true,
        fee: '1',
        minTransfer: '0.0001',
        type: '1'
      }
    ],
    withdraw: { fee: undefined, percentage: undefined },
    deposit: { fee: undefined, percentage: undefined },
    networks: {
      ERC20: {
        withdraw: { fee: 30, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      },
      TRC20: {
        withdraw: { fee: 1, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      },
      BSC: {
        withdraw: { fee: 1, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      },
      HT: {
        withdraw: { fee: 1, percentage: undefined },
        deposit: { fee: undefined, percentage: undefined }
      }
    }
  },
  ...
}
2022-11-24T23:24:27.075Z iteration 1 passed in 730 ms
```

### CODE REVIEW

File: js/lbank2.js [hunk 1]
```diff
@@ -2064,6 +2069,217 @@ module.exports = class lbank2 extends Exchange {
...
+    async fetchDepositWithdrawFees (codes = undefined, params = {}) {
+        /**
+         * @method
+         * @name lbank2#fetchDepositWithdrawFees
+         * @description when using private endpoint, only returns information for currencies with non-zero balance, use public method by specifying this.options['fetchDepositWithdrawFees']['method'] = 'fetchPublicDepositWithdrawFees'
+         * @param {[string]|undefined} codes array of unified currency codes
+         * @param {object} params extra parameters specific to the lbank2 api endpoint
+         * @returns {object} a list of [fee structures]{@link https://docs.ccxt.com/en/latest/manual.html#fee-structure}
+         */
+        await this.loadMarkets ();
```
Comment (carlosmiei) [inline js/lbank2.js:2081]: Again, shouldn't we fill this up with the "default" network value? 
image

Comment (samgermain) [reply js/lbank2.js:2081]: There's an `isDefault` key, which is checked for, but USDT has no key where `isDefault` is set to true

Comment (samgermain) [reply js/lbank2.js:2081]: The default is set for withdraw fees, deposit fees aren't provided

Comment (carlosmiei) [reply js/lbank2.js:2081]: Yeah, it was at the end of the response, so I didn't notice it; thanks!

**Comment by carlosmiei** [comment]
LGTM ✅ 

```
p lbank2 fetchDepositWithdrawFees '["USDT"]'
Python v3.9.7
CCXT v2.2.36
lbank2.fetchDepositWithdrawFees(['USDT'])
{'USDT': {'deposit': {'fee': None, 'percentage': None},
          'info': {'assetAmt': '10',
                   'coin': 'usdt',
                   'freezeAmt': '0',
                   'networkList': [{'coin': 'usdt',
                                    'feeAssetCode': 'usdt',
                                    'isDefault': False,
                                    'minDeposit': '20',
                                    'minLimit': '0.0001',
                                    'name': 'erc20',
                                    'network': 'eth',
                                    'type': '1',
                                    'withdrawFee': '20',
                                    'withdrawFeeRate': '',
                                    'withdrawMin': '20'},
                                   {'coin': 'usdt',
                                    'isDefault': False,
                                    'name': 'omni',
                                    'network': 'btc'},
                                   {'coin': 'usdt',
                                    'feeAssetCode': 'usdt',
                                    'isDefault': False,
                                    'minDeposit': '0.1',
                                    'minLimit': '0.0001',
                                    'name': 'trc20',
                                    'network': 'trx',
                                    'type': '1',
                                    'withdrawFee': '1',
                                    'withdrawFeeRate': '',
                                    'withdrawMin': '2'},
                                   {'coin': 'usdt',
                                    'feeAssetCode': 'usdt',
                                    'isDefault': False,
                                    'minDeposit': '5',
                                    'minLimit': '0.0001',
                                    'name': 'asa',
                                    'network': 'algo',
                                    'type': '1',
                                    'withdrawFee': '5',
                                    'withdrawFeeRate': '',
                                    'withdrawMin': '1E+1'},
                                   {'coin': 'usdt',
                                    'feeAssetCode': 'usdt',
                                    'isDefault': False,
                                    'minDeposit': '0.0001',
                                    'minLimit': '0.0001',
                                    'name': 'bep20(bsc)',
                                    'network': 'bsc',
                                    'type': '1',
                                    'withdrawFee': '1',
                                    'withdrawFeeRate': '',
                                    'withdrawMin': '10'},
                                   {'coin': 'usdt',
                                    'feeAssetCode': 'usdt',
                                    'isDefault': False,
                                    'minDeposit': '0.1',
                                    'minLimit': '0.0001',
                                    'name': 'heco',
                                    'network': 'heco',
                                    'type': '1',
                                    'withdrawFee': '1',
                                    'withdrawFeeRate': '',
                                    'withdrawMin': '2'},
                                   {'coin': 'usdt',
                                    'feeAssetCode': 'usdt',
                                    'isDefault': True,
                                    'minDeposit': '1',
                                    'minLimit': '0.0001',
                                    'name': 'matic',
                                    'network': 'matic',
                                    'type': '1',
                                    'withdrawFee': '1',
                                    'withdrawFeeRate': '',
                                    'withdrawMin': '1'}],
                   'usableAmt': '10'},
          'networks': {'ASA': {'deposit': {'fee': None, 'percentage': None},
                               'withdraw': {'fee': 5.0, 'percentage': None}},
                       'BSC': {'deposit': {'fee': None, 'percentage': None},
                               'withdraw': {'fee': 1.0, 'percentage': None}},
                       'ERC20': {'deposit': {'fee': None, 'percentage': None},
                                 'withdraw': {'fee': 20.0, 'percentage': None}},
                       'HT': {'deposit': {'fee': None, 'percentage': None},
                              'withdraw': {'fee': 1.0, 'percentage': None}},
                       'MATIC': {'deposit': {'fee': None, 'percentage': None},
                                 'withdraw': {'fee': 1.0, 'percentage': None}},
                       'TRC20': {'deposit': {'fee': None, 'percentage': None},
                                 'withdraw': {'fee': 1.0, 'percentage': None}}},
          'withdraw': {'fee': 1.0, 'percentage': None}}}
```

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**